gapis/service/memory_box/box.go

```diff
@@ -248,15 +255,44 @@ func Box(ctx context.Context, d *memory.Decoder, t *types.Type) (*Value, error)
 		}
 	case *types.Type_Pseudonym:
 		if elem, ok := types.TryGetType(t.Pseudonym.Underlying); ok {
-			return Box(ctx, d, elem)
+			b, err := Box(ctx, d, elem, p, rc)
+			if err != nil {
+				return nil, err
+			}
+
+			reflType, err := types.GetReflectedType(typeId)
+			if err != nil {
+				return nil, err
+			}
+
+			if reflType.Implements(tyLinkable) {
+				// reflectively convert the underlying value back to what its
+				// api type should have been, then attempt to link through it.
+				var v reflect.Value
+				switch vv := b.Val.(*Value_Pod).Pod.Val.(type) {
+				case (*pod.Value_Uint64):
+					v = reflect.ValueOf(vv.Uint64)
+				case (*pod.Value_Uint32):
+					v = reflect.ValueOf(vv.Uint32)
+				default:
+					return nil, log.Errf(ctx, nil, "Cannot handle pseudonym underlying type")
+				}
+
+				linker := v.Convert(reflType).Interface().(path.Linker)
```

Comment:
I think a slightly less convoluted way to do this is:
```
v := reflect.New(reflType)
switch vv := b.Val.(*Value_Pod).Pod.Val.(type) {
  case (*pod.Value_Uint64):
    v.Elem().SetInt(vv.Uint64)
  ...
}
if res, err := v.Interface().(path.Linker).Link(...); err == nil {
}
```
... at least I think that should work...

Reply:
Thanks, done. I was initially not thrilled about poking values into the guts like this, but it does simplify a bit.